CONTRIBUTING.md

```diff
@@ -14,13 +14,20 @@ If you find a bug in the code or a mistake in the [documentation](https://vega.g
 
 See our [issue](.github/ISSUE_TEMPLATE.md) and [pull request](.github/PULL_REQUEST_TEMPLATE.md) templates for more information.
 
-### Looking for a Task to Contribute
+## Design principles
```

Comment:
Some of these items you listed down here feels more like contract to users (regarding versioning)) more than design principles. 

What we had in https://medium.com/hci-design-at-uw/introducing-vega-lite-438f9215f09e feels more like the genuine design principles of the language to me.

Reply:
I think a mix of both are good in the contributing guidelines. Maybe design principles is not as good as a term as development principles or development goals.

Reply:
I added "Favor composition over templates" as it was missing. 

"Leverage Vega’s performance, flexibility across platforms, and expressivity." doesn't make too much sense in this context, I think.